### PR TITLE
Ginkgo: Refactor Kubectl.ExecPodCmd

### PR DESCRIPTION
**Pull request by eloycoto**
ExecPodCmd function uses a unique buffer instead of the global one. The
execution was not saved on log, and the output was not the standard
ResCmd. With this change, all ExecPodCmd calls are using the standard
way.

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

### CODE REVIEW

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by ianvernon** [comment]
@eloycoto can you triage the test failure to see if it was caused by the changes in this PR? I will review if it is unrelated / once tests pass - thanks!

File: test/k8sT/Nightly.go [hunk 1]
```diff
@@ -350,15 +352,17 @@ var _ = Describe("NightlyExamples", func() {
...
 				helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIP)))
-			Expect(err).Should(BeNil())
+			res.ExpectSuccess("Cannot curl to %q from %q", clusterIP, appPods[helpers.App2])
 
-			_, err = kubectl.ExecPodCmd(
+			res = kubectl.ExecPodCmd(
 				helpers.DefaultNamespace, appPods[helpers.App3],
 				helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIP)))
-			Expect(err).Should(HaveOccurred())
+			res.ExpectFail("Can curl to %q from %q and it should",
```
Comment (ianvernon) [inline test/k8sT/Nightly.go]: and it should --> (it shouldn't)

File: test/k8sT/Policies.go [hunk 2]
```diff
@@ -198,15 +198,15 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
 			trace.ExpectSuccess(trace.CombineOutput().String())
 			Expect(trace.Output().String()).Should(ContainSubstring("Final verdict: DENIED"))
 
-			_, err = kubectl.ExecPodCmd(
+			res := kubectl.ExecPodCmd(
 				helpers.DefaultNamespace, appPods[helpers.App2],
-				fmt.Sprintf("curl http://%s/public", clusterIP))
-			Expect(err).Should(BeNil())
+				helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIP)))
+			res.ExpectSuccess("App2 %q cannot curl clusterIP %q", appPods[helpers.App2], clusterIP)
```
Comment (ianvernon) [inline test/k8sT/Policies.go]: Remove "App2", it's redundant.

File: test/k8sT/Policies.go [hunk 3]
```diff
@@ -198,15 +198,15 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
...
 
-			_, err = kubectl.ExecPodCmd(
+			res = kubectl.ExecPodCmd(
 				helpers.DefaultNamespace, appPods[helpers.App3],
 				helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIP)))
-			Expect(err).Should(HaveOccurred())
+			res.ExpectFail("App3 %q can curl to %q", appPods[helpers.App3], clusterIP)
```
Comment (ianvernon) [inline test/k8sT/Policies.go]: Remove "App3", it's redundant.

File: test/k8sT/Policies.go [hunk 4]
```diff
@@ -229,25 +229,26 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
 
 			appPods = helpers.GetAppPods(apps, helpers.DefaultNamespace, kubectl, "id")
 
-			_, err = kubectl.ExecPodCmd(
+			res = kubectl.ExecPodCmd(
 				helpers.DefaultNamespace, appPods[helpers.App2],
-				fmt.Sprintf("curl http://%s/public", clusterIP))
-			Expect(err).Should(BeNil())
+				helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIP)))
+			res.ExpectSuccess("App2 %q cannot curl clusterIP %q", appPods[helpers.App2], clusterIP)
```
Comment (ianvernon) [inline test/k8sT/Policies.go]: Remove "App2", it's redundant.

File: test/k8sT/Policies.go [hunk 5]
```diff
@@ -229,25 +229,26 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
...
 
-			msg, err := kubectl.ExecPodCmd(
+			res = kubectl.ExecPodCmd(
 				helpers.DefaultNamespace, appPods[helpers.App2],
-				fmt.Sprintf("curl --fail -s http://%s/private", clusterIP))
-			Expect(err).Should(HaveOccurred(), msg)
+				helpers.CurlFail(fmt.Sprintf("http://%s/private", clusterIP)))
+			res.ExpectFail("App2 %q cannot curl clusterIP %q private",
```
Comment (ianvernon) [inline test/k8sT/Policies.go]: Remove "App2", it's redundant.

File: test/k8sT/Policies.go [hunk 6]
```diff
@@ -306,30 +308,31 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
...
 			Expect(err).To(BeNil(), "Cannot get service on %q namespace", namespace)
 			appPods := helpers.GetAppPods(apps, namespace, kubectl, "id")
-			_, err = kubectl.ExecPodCmd(
+
+			res := kubectl.ExecPodCmd(
 				namespace, appPods[helpers.App2],
-				fmt.Sprintf("curl http://%s/public", clusterIP))
-			Expect(err).Should(BeNil())
+				helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIP)))
+			res.ExpectSuccess("App2 %q cannot curl clusterIP %q", appPods[helpers.App2], clusterIP)
```
Comment (ianvernon) [inline test/k8sT/Policies.go]: Remove "App2", it's redundant.

File: test/k8sT/Policies.go [hunk 7]
```diff
@@ -257,10 +258,11 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
 			err = helpers.WaitUntilEndpointUpdates(ciliumPod, eps, 4, kubectl)
 			Expect(err).Should(BeNil())
 
-			_, err = kubectl.ExecPodCmd(
+			res = kubectl.ExecPodCmd(
 				helpers.DefaultNamespace, appPods[helpers.App3],
-				fmt.Sprintf("curl --fail -s http://%s/public", clusterIP))
-			Expect(err).Should(BeNil())
+				helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIP)))
+			res.ExpectSuccess("App3 %q cannot curl to %q public", appPods[helpers.App3], clusterIP)
```
Comment (ianvernon) [inline test/k8sT/Policies.go]: Remove "App3", it's redundant.

File: test/k8sT/Policies.go [hunk 8]
```diff
@@ -229,25 +229,26 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
...
+				helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIP)))
+			res.ExpectFail("App3 %q can curl to %q", appPods[helpers.App3], clusterIP)
 
-			msg, err = kubectl.ExecPodCmd(
+			res = kubectl.ExecPodCmd(
 				helpers.DefaultNamespace, appPods[helpers.App3],
-				fmt.Sprintf("curl --fail -s http://%s/private", clusterIP))
-			Expect(err).Should(HaveOccurred(), msg)
+				helpers.CurlFail(fmt.Sprintf("http://%s/private", clusterIP)))
+			res.ExpectFail("App3 %q can curl to %q private", appPods[helpers.App3], clusterIP)
```
Comment (ianvernon) [inline test/k8sT/Policies.go]: Remove "App3", it's redundant.

File: test/k8sT/Policies.go [hunk 9]
```diff
@@ -229,25 +229,26 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
...
+			res.ExpectFail("App2 %q cannot curl clusterIP %q private",
+				appPods[helpers.App2], clusterIP)
 
-			_, err = kubectl.ExecPodCmd(
+			res = kubectl.ExecPodCmd(
 				helpers.DefaultNamespace, appPods[helpers.App3],
-				fmt.Sprintf("curl --fail -s http://%s/public", clusterIP))
-			Expect(err).Should(HaveOccurred())
+				helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIP)))
+			res.ExpectFail("App3 %q can curl to %q", appPods[helpers.App3], clusterIP)
```
Comment (ianvernon) [inline test/k8sT/Policies.go]: Remove "App3", it's redundant.

File: test/k8sT/Policies.go [hunk 10]
```diff
@@ -306,30 +308,31 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
...
-			Expect(err).Should(BeNil())
+				helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIP)))
+			res.ExpectSuccess("App2 %q cannot curl clusterIP %q", appPods[helpers.App2], clusterIP)
 
-			_, err = kubectl.ExecPodCmd(
+			res = kubectl.ExecPodCmd(
 				namespace, appPods[helpers.App3],
 				helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIP)))
-			Expect(err).Should(HaveOccurred())
+			res.ExpectFail("App3 %q can curl to %q", appPods[helpers.App3], clusterIP)
```
Comment (ianvernon) [inline test/k8sT/Policies.go]: Remove "App3", it's redundant.

File: test/k8sT/Policies.go [hunk 11]
```diff
@@ -306,30 +308,31 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
...
-
 			appPods = helpers.GetAppPods(apps, helpers.DefaultNamespace, kubectl, "id")
-			_, err = kubectl.ExecPodCmd(
+
+			res = kubectl.ExecPodCmd(
 				helpers.DefaultNamespace, appPods[helpers.App2],
-				fmt.Sprintf("curl http://%s/public", clusterIP))
-			Expect(err).Should(BeNil())
+				helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIP)))
+			res.ExpectSuccess("App2 %q cannot curl clusterIP %q", appPods[helpers.App2], clusterIP)
```
Comment (ianvernon) [inline test/k8sT/Policies.go]: Remove "App2", it's redundant.

File: test/k8sT/Policies.go [hunk 12]
```diff
@@ -306,30 +308,31 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
...
 
-			_, err = kubectl.ExecPodCmd(
+			res = kubectl.ExecPodCmd(
 				helpers.DefaultNamespace, appPods[helpers.App3],
 				helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIP)))
-			Expect(err).Should(HaveOccurred())
+			res.ExpectFail("App3 %q can curl to %q", appPods[helpers.App3], clusterIP)
```
Comment (ianvernon) [inline test/k8sT/Policies.go]: Remove "App3", it's redundant.

File: test/k8sT/Policies.go [hunk 13]
```diff
@@ -354,23 +357,25 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
...
 			Expect(err).To(BeNil(), "Cannot get service on %q namespace", namespace)
 			appPods := helpers.GetAppPods(apps, namespace, kubectl, "id")
-			_, err = kubectl.ExecPodCmd(
+
+			res := kubectl.ExecPodCmd(
 				namespace, appPods[helpers.App2],
-				fmt.Sprintf("curl http://%s/public", clusterIP))
-			Expect(err).Should(BeNil())
+				helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIP)))
+			res.ExpectSuccess("App2 %q cannot curl clusterIP %q", appPods[helpers.App2], clusterIP)
```
Comment (ianvernon) [inline test/k8sT/Policies.go]: Remove "App2", it's redundant.

File: test/k8sT/Policies.go [hunk 14]
```diff
@@ -354,23 +357,25 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
...
 
-			_, err = kubectl.ExecPodCmd(
+			res = kubectl.ExecPodCmd(
 				namespace, appPods[helpers.App3],
-				fmt.Sprintf("curl http://%s/public", clusterIP))
-			Expect(err).Should(BeNil())
+				helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIP)))
+			res.ExpectFail("App3 %q can curl to %q", appPods[helpers.App3], clusterIP)
```
Comment (ianvernon) [inline test/k8sT/Policies.go]: Remove "App3", it's redundant.

**Review by ianvernon — COMMENTED** [review]
Overall this is a great change - uniformity across the helper functions is awesome :)

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Review by ianvernon — APPROVED** [review]

**Pull request merged**